osu.Game/Skinning/LegacySkin.cs

```diff
@@ -442,6 +388,84 @@ protected override void ParseConfigurationStream(Stream stream)
             return null;
         }
 
+        private static DefaultSkinComponentsContainer? createDefaultHUDComponents(SkinComponentsContainerLookup containerLookup)
+        {
+            switch (containerLookup.Ruleset?.ShortName)
+            {
+                case null:
+                {
+                    return new DefaultSkinComponentsContainer(container =>
+                    {
+                        var score = container.OfType<LegacyScoreCounter>().FirstOrDefault();
+                        var accuracy = container.OfType<GameplayAccuracyCounter>().FirstOrDefault();
+
+                        if (score != null && accuracy != null)
+                        {
+                            accuracy.Y = container.ToLocalSpace(score.ScreenSpaceDrawQuad.BottomRight).Y;
+                        }
+
+                        var songProgress = container.OfType<LegacySongProgress>().FirstOrDefault();
+
+                        if (songProgress != null && accuracy != null)
+                        {
+                            songProgress.Anchor = Anchor.TopRight;
+                            songProgress.Origin = Anchor.CentreRight;
+                            songProgress.X = -accuracy.ScreenSpaceDeltaToParentSpace(accuracy.ScreenSpaceDrawQuad.Size).X - 18;
+                            songProgress.Y = container.ToLocalSpace(accuracy.ScreenSpaceDrawQuad.TopLeft).Y + (accuracy.ScreenSpaceDeltaToParentSpace(accuracy.ScreenSpaceDrawQuad.Size).Y / 2);
+                        }
+
+                        var hitError = container.OfType<HitErrorMeter>().FirstOrDefault();
+
+                        if (hitError != null)
+                        {
+                            hitError.Anchor = Anchor.BottomCentre;
+                            hitError.Origin = Anchor.CentreLeft;
+                            hitError.Rotation = -90;
+                        }
+                    })
+                    {
+                        Children = new Drawable[]
+                        {
+                            new LegacyComboCounter(),
+                            new LegacyScoreCounter(),
+                            new LegacyAccuracyCounter(),
+                            new LegacySongProgress(),
+                            new LegacyHealthDisplay(),
+                            new BarHitErrorMeter(),
+                        }
+                    };
+                }
+
+                case @"osu":
+                case @"fruits":
```

Comment:
will it better to just show the `LegacyKeyCounterDisplay` at first and hide the `LegacyKeyCounterDisplay()` at mania, and taiko ruleset?

Reply:
I intentionally pushed this implementation over what you had. This is a bridge that will eventually have to be crossed anyway with differing implementations of legacy combo counters in osu! / the other rulesets.

Reply:
Can we remain this https://github.com/ppy/osu/blob/0306ef4096aeb8aafd1eb49c8afefcd338a62f05/osu.Game/Skinning/LegacySkin.cs#L351-L444
and do this to hide the `LegacyKeyCounterDisplay` in mania ruleset
```diff 
:100644 100644 73c521b2ed 0000000000 M  osu.Game.Rulesets.Mania/Skinning/Legacy/ManiaLegacySkinTransformer.cs

diff --git a/osu.Game.Rulesets.Mania/Skinning/Legacy/ManiaLegacySkinTransformer.cs b/osu.Game.Rulesets.Mania/Skinning/Legacy/ManiaLegacySkinTransformer.cs
index 73c521b2ed..3680306f7d 100644
--- a/osu.Game.Rulesets.Mania/Skinning/Legacy/ManiaLegacySkinTransformer.cs
+++ b/osu.Game.Rulesets.Mania/Skinning/Legacy/ManiaLegacySkinTransformer.cs
@@ -5,9 +5,11 @@

 using System;
 using System.Collections.Generic;
+using System.Linq;
 using osu.Framework.Audio.Sample;
 using osu.Framework.Bindables;
 using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
 using osu.Game.Audio;
 using osu.Game.Beatmaps;
 using osu.Game.Rulesets.Mania.Beatmaps;
@@ -76,6 +78,22 @@ public ManiaLegacySkinTransformer(ISkin skin, IBeatmap beatmap)

         public override Drawable GetDrawableComponent(ISkinComponentLookup lookup)
         {
+            if (lookup is SkinComponentsContainerLookup containerLookup)
+            {
+                switch (containerLookup.Target)
+                {
+                    case SkinComponentsContainerLookup.TargetArea.MainHUDComponents:
+                        var components = base.GetDrawableComponent(lookup) as Container;
+
+                        foreach (var keyCounterDisplay in components?.OfType<LegacyKeyCounterDisplay>())
+                        {
+                            keyCounterDisplay.Hide();
+                        }
+
+                        return components;
+                }
+            }
+
             switch (lookup)
             {
                 case GameplaySkinComponentLookup<HitResult> resultComponent:
```

Reply:
No. Fixing a unit test is an even more flimsy reasoning to keep your change. https://github.com/ppy/osu/pull/29027/commits/12a9086aa31127b4b81f5a02cbcfe190b4159b5b is how you _actually_ fix the unit test.

Unit test fixes should never be given as a rationale for an actual behavioural change in the code such as this.